### ISSUE TITLE
Default social provider buttons don’t work with basic setup

### ISSUE THREAD

**Issue by YahngSungho** [aws-amplify/amplify-js#4269]
**Describe the bug**

![socialProvider1](https://user-images.githubusercontent.com/29212791/67633255-2c90ba80-f8f1-11e9-8362-06b0bb0915da.gif)

also in console:
``
NotAuthorizedException: Token is not from a supported provider of this identity pool.
``

But ‘Sign in with AWS’ seems to work.

![2019-10-27 18 52 11 Signin - Chrome](https://user-images.githubusercontent.com/29212791/67633259-3ca89a00-f8f1-11e9-98d5-62fe2b914c7b.png)

One thing to consider is that I mistyped the information to the federated prop. That's the only difference I can think of. But I'm sure they're right.

I don't know if it's relevant, but one more weird thing is that Authenticator Comp doesn't work either. To be clear, it only show ‘Loading...’ And if I hide loading, It doesn't show anything. So I use only withAuthenticator HOC.
(But Authenticator Comp doesn't work too when I don't provide federated prop, so it might be another problem)

So basically my code is this:
```
Amplify.configure(awsconfig)
const federated = {
  google_client_id: '<google-id>',
  facebook_app_id: '<fb-id>',
  amazon_client_id: ''
}

export default withAuthenticator(App, true, [], federated)
```

I think there is nothing special in my code. I just did a basic setup. Maybe something wrong with the amplify installation or configuration, but I don't know where to look.

Also I don't know if it's relevant, it's my `aws-export.js`
```
// WARNING: DO NOT EDIT. This file is automatically generated by AWS Amplify. It will be overwritten.

const awsmobile = {
    "aws_project_region": "us-east-1",
    "aws_cognito_identity_pool_id": "<idp-id>",
    "aws_cognito_region": "us-east-1",
    "aws_user_pools_id": "<up-id>",
    "aws_user_pools_web_client_id": "<upwc-id>",
    "oauth": {
        "domain": "<oa-domain>",
        "scope": [
            "phone",
            "email",
            "openid",
            "profile",
            "aws.cognito.signin.user.admin"
        ],
        "redirectSignIn": "http://localhost:5000/",
        "redirectSignOut": "http://localhost:5000/",
        "responseType": "code"
    },
    "federationTarget": "COGNITO_USER_POOLS"
};


export default awsmobile;
``` 
**Desktop (please complete the following information):**

- OS: Window 7
- Browser: chrome
- Version: latest

**Amplify context**
- Framework: React
- Versions of Amplify you are using: latest

**Comment by haruharuharuby** [comment]
Hi,  @sammartinez (CC: @YahngSungho )

I got the same issue with a super simple code. I did not change code mostly in this sample.
https://docs.amplify.aws/lib/auth/social/q/platform/js#full-react-sample

And I only use the Cognito Hosted UI as Social Provider.

I did debug the problem.  Do you have some wonder about package dependencies?
In my case, I did debug as follows.

1. The NotAuthorizedException was raised from the GetIdCommand this command exists in amplify-core.
https://github.com/aws-amplify/amplify-js/blob/master/packages/core/src/Credentials.ts#L239
This is a method from aws-sdk-v3.   AWS SDK v3 is currently beta yet.  

2. With debugging on Chrome, 
The token endpoint of Cognito was successfully called.
I confirmed the accessToken internally.
But later it's in the above operation, Calling Cognito-identity endpoint was failed.
This is from the GetIdCommand.

<img width="1164" alt="chrome-debug-token-ep-success" src="https://user-images.githubusercontent.com/10077078/81551439-25f3fd00-9382-11ea-9f7f-ec0cef806866.png">

<img width="1406" alt="chrome-debug-cognito-idp-fail" src="https://user-images.githubusercontent.com/10077078/81551448-28565700-9382-11ea-8e8b-4f10b5482c01.png">

 I can implement OAuth flow manually, But of course, it's better to simpler implementation.
If I can check something, Please let me know.

 I'm happy if this issue will be handled as fast as possible.

**Comment by ericclemmons** [comment]
@YahngSungho @haruharuharuby Can you adjust your code to use `Auth.federatedSignIn`, according to:
> https://docs.amplify.aws/lib/auth/social/q/platform/js#setup-frontend

That should work out-of-the-box with the least amount of custom code to validate that federated sign-in works & is configured correctly. (I want to confirm that @YahngSungho & @haruharuharuby are experiencing the same thing)

You're doing the correct thing by signing in with Hosted UI first to validate your config (this URL can be found via `amplify status`!), then `Auth.federatedSignIn()` from within your app, then finally individual `Auth.federatedSignIn({ provider: 'Google' })` buttons.

**Comment by haruharuharuby** [comment]
@ericclemmons (CC: @YahngSungho )

I could resolve it myself in my case.
My problem resulted from a shortage of Cognito Identity Provider.
I needed to add UserPoolId and ClientId to IdentityProvider.
Because I configured the aws-exports.js to specify existed UserPool and Identity Provider and I added a new Client on only UserPool configuration.

I compared UserPool and Cognito Idp configuration between existing one and newly deployed one which published by calling "amplify add auth".

It may fix if @YahngSungho's configuration is the same.
<img width="1437" alt="screenshot" src="https://user-images.githubusercontent.com/10077078/82049882-6ff91d80-96b7-11ea-87bd-05d123b03cd5.png">
